src/components/ScenarioParameters/components/ScenarioParametersInputs/components/TableExportDialog/TableExportDialog.js

```diff
@@ -0,0 +1,133 @@
+  };
+  const selectFileType = (event) => {
+    setFileType(event.target.value);
+  };
+  const confirmExport = () => {
+    const exportedFileName = `${fileName}.${fileType}`;
+    onExport(exportedFileName);
+  };
+
+  return (
+    <Dialog onClose={onClose} open={open} data-cy="table-export-dialog">
+      <DialogTitle>{labels.title}</DialogTitle>
+      <DialogContent sx={{ paddingBottom: '0' }}>
+        <Typography>{labels.label}</Typography>
+        <Grid container spacing={2} direction="column" justifyContent="center" alignItems="flex-start">
+          <Grid item data-cy="table-export-file-type-container">
+            <FormControl size="small" sx={{ marginTop: 2, width: 220 }}>
+              <InputLabel id="table-export-file-type-select-label">{labels.fileTypeSelectLabel}</InputLabel>
+              <Select
+                data-cy="table-export-file-type-select"
+                labelId="table-export-file-type-select-label"
+                id="table-export-file-type-select"
+                value={fileType}
+                label={labels.fileTypeSelectLabel}
+                onChange={selectFileType}
+              >
+                {ALLOWED_FILE_TYPES.map((allowedFileType) => (
+                  <MenuItem
+                    key={allowedFileType.extension}
+                    value={allowedFileType.extension}
+                    data-cy={`table-export-file-type-select-option-${allowedFileType.extension}`}
+                  >
+                    {allowedFileType.label}
+                  </MenuItem>
+                ))}
+              </Select>
+            </FormControl>
+          </Grid>
+          <Grid item>
+            <TextField
+              id="table-export-file-name-input"
+              data-cy="table-export-file-name-input"
+              label={labels.fileNameInputLabel}
+              variant="outlined"
+              value={fileName}
+              onChange={selectFileName}
+              size="small"
+              InputProps={{
+                endAdornment: <InputAdornment position="end">{'.' + fileType}</InputAdornment>,
+              }}
+              sx={{ width: 220 }}
+            />
+          </Grid>
+        </Grid>
+      </DialogContent>
+      <DialogActions>
+        <Button data-cy="table-export-file-cancel-button" onClick={onClose}>
+          {labels.cancel}
+        </Button>
+        <Button data-cy="table-export-file-confirm-button" onClick={confirmExport} variant="contained" autoFocus>
+          {labels.export}
+        </Button>
+      </DialogActions>
+    </Dialog>
+  );
+};
+
+TableExportDialog.propTypes = {
+  defaultFileName: PropTypes.string,
+  labels: PropTypes.shape({
+    cancel: PropTypes.string.isRequired,
+    export: PropTypes.string.isRequired,
+    fileNameInputLabel: PropTypes.string.isRequired,
+    fileTypeSelectLabel: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    label: PropTypes.string.isRequired,
```

Comment:
Is it necessary to make these labels required ? They seem quite generic to me :slightly_smiling_face:

Reply:
I'd say this is necessary to inform developers that when providing the `labels` prop, all these keys must be present, otherwise some labels will be missing. IIRC, this behavior comes from the way React `defaultProps` works:
- if the prop `labels` is undefined, then its value will be set to the one of `TableExportDialog.defaultProps.labels` (with all default labels we defined)
- if the prop `labels` is defined, but some keys are missing, they won't be copied from `TableExportDialog.defaultProps.labels` leading to some text being missing

A work-around for this would be to "patch" the `labels` prop received with a dict containing all default values, but the result is less readable, and I'm not sure it is very conventional :sweat_smile: (see [this PR](https://github.com/Cosmo-Tech/webapp-component-ui/pull/313/files) for an example of "default values patching" :slightly_smiling_face: )

Reply:
Note: another work-around would be to add fallback values with the `??` operator every time we use the `labels` prop, but it seems less readable and harder to maintain than a single dict of all default values

Reply:
I agree, it will be messy with ?? operators everywhere and patching can be useful when there are lots of labels. +1 to keep props required, in my opinion, it won't be changed a lot (not easy to find an alternative to _Cancel_ :rofl: )